File: problem0003/java/Problem0003.java
/**
 * Problem: http://projecteuler.net/index.php?section=problems&id=3
 * Author: Elie De Brauwer <elie @ de-brauwer.be>
 * License: Simplified BSD
 * Date: December 12, 2010
 */

import java.math.BigInteger;

class Problem0003{
    public static void main(String [] args){
	BigInteger val = new BigInteger("600851475143");
	BigInteger cur = new BigInteger("2");
	cur = cur.nextProbablePrime();
	while(val.compareTo(cur.multiply(cur)) != -1){
	    while(val.mod(cur) == BigInteger.ZERO){
		val = val.divide(cur);
		System.out.println(cur.toString());
	    }
	    cur = cur.nextProbablePrime();
	}
	System.out.println(val.toString());
    }
}

File: problem0003/haskell/problem0003.hs
{-
Problem: http://projecteuler.net/index.php?section=problems&id=3
Author: Elie De Brauwer <elie @ de-brauwer.be>
License: Simplified BSD
Date: December 14, 2010
-}

{- n = candidate divisor, a = divident -}
factor :: Integer -> Integer -> [Integer]
factor _ 1 = []
factor n a = if (a `mod` n == 0) then n : factor n (a `div` n) else factor (n+1) a

problem0003 :: [Integer]
problem0003 = factor 2 600851475143

main = print problem0003